### PR TITLE
Update transaction tests to sepolia

### LINKED ISSUE #544: goerli depreceation
Any tests and examples that currently use Goerli should use another test network, like Sepolia. 

1. Enter the rpc directory
2. rename .env.template to .env.testnet, and enter the Sepolia endpoint
3. go test ./... -env testnet

You will see a bunch of tests that currently depend on Goerli, but should be updated to depend on Sepolia instead.

Free RPC endpoints https://data.voyager.online/

### PR DESCRIPTION
**Pull request by AryanGodara**
Partially resolves #544

### CODE REVIEW

File: rpc/transaction_test.go [hunk 1]
```diff
@@ -23,54 +23,31 @@ func TestTransactionByHash(t *testing.T) {
...
 		},
+		SenderAddress: utils.TestHexToFelt(t, "0x5fd4befee268bf6880f955875cbed3ade8346b1f1e149cc87b317e62b6db569"),
 	}
 
 	testSet := map[string][]testSetType{
 		"mock": {
 			{
-				TxHash:      utils.TestHexToFelt(t, "0x1779df1c6de5136ad2620f704b645e9cbd554b57d37f08a06ea60142269c5a5"),
-				ExpectedTxn: InvokeTxnV1example,
+				TxHash:      utils.TestHexToFelt(t, "0xd109474cd037bad60a87ba0ccf3023d5f2d1cd45220c62091d41a614d38eda"),
```
Comment (rianhughes) [inline rpc/transaction_test.go:43]: For the mock, you'll need to update `mock_starknet_getTransactionByHash` to return the declare transaction

File: rpc/transaction_test.go [hunk 2]
```diff
@@ -118,28 +95,70 @@ func TestTransactionByBlockIdAndIndex(t *testing.T) {
...
+		Tip: "0x0",
 	}
 
 	testSet := map[string][]testSetType{
 		"mock": {
 			{
 				BlockID:     WithBlockNumber(300000),
 				Index:       0,
-				ExpectedTxn: InvokeTxnV1example,
+				ExpectedTxn: InvokeTxnV3example,
```
Comment (rianhughes) [inline rpc/transaction_test.go:154]: Same for mock_starknet_getTransactionByBlockIdAndIndex

File: rpc/transaction_test.go (hunk 1 above)
Comment (rianhughes) [reply rpc/transaction_test.go:43]: Mock tests actually call these functions under the hood

File: rpc/transaction_test.go [hunk 3]
```diff
@@ -170,41 +190,51 @@ func TestTransactionReceipt(t *testing.T) {
...
+					utils.TestHexToFelt(t, "0x40851db0ebaebb9f8a18eda25005c050793f2a69e9e7d1f44bc133752898918"),
+				},
+				Keys: []*felt.Felt{
+					utils.TestHexToFelt(t, "0x243d436e1f7cea085aaa42834975488029b1ebf67cea1d2e86f7de58e7d34a3"),
 				},
-				Keys: []*felt.Felt{utils.TestHexToFelt(t, "0x99cd8bde557814842a3121e8ddfd433a539b8c9f14bf31ebf108d12e6196e9")},
 			},
 			{
-				FromAddress: utils.TestHexToFelt(t, "0x33830ce413e4c096eef81b5e6ffa9b9f5d963f57b8cd63c9ae4c839c383c1a6"),
+				FromAddress: utils.TestHexToFelt(t, "0x243d436e1f7cea085aaa42834975488029b1ebf67cea1d2e86f7de58e7d34a3"),
```
Comment (rianhughes) [inline rpc/transaction_test.go]: The second event data is incorrect

File: rpc/transaction_test.go [hunk 4]
```diff
@@ -170,41 +190,51 @@ func TestTransactionReceipt(t *testing.T) {
...
+					utils.TestHexToFelt(t, "0x16409a78a10b00"),
+					utils.TestHexToFelt(t, "0x0"),
+				},
+				Keys: []*felt.Felt{
+					utils.TestHexToFelt(t, "0x15bd0500dc9d7e69ab9577f73a8d753e8761bed10f25ba0f124254dc4edb8b4"),
 				},
-				Keys: []*felt.Felt{utils.TestHexToFelt(t, "0xf806f71b19e4744968b37e3fb288e61309ab33a782ea9d11e18f67a1fbb110")},
 			},
 		},
 		ExecutionResources: ExecutionResources{
```
Comment (rianhughes) [inline rpc/transaction_test.go:230]: we'll also need to insert the data_availability field

File: rpc/transaction_test.go [hunk 5]
```diff
@@ -170,41 +190,51 @@ func TestTransactionReceipt(t *testing.T) {
...
-				Steps:          217182,
-				MemoryHoles:    6644,
-				PedersenApps:   2142,
-				RangeCheckApps: 8867,
-				BitwiseApps:    900,
-				ECDSAApps:      1,
+				Steps:          5774,
+				PedersenApps:   24,
+				ECOPApps:       3,
+				RangeCheckApps: 152,
```
Comment (rianhughes) [inline rpc/transaction_test.go:235]: We should also update the test to require the entire receipt to be equal to what we get from the full node:
require.Equal(t, txReceiptWithBlockInfo, test.ExpectedResp)

File: rpc/types_transaction_receipt.go [hunk 6]
```diff
@@ -25,6 +25,8 @@ const (
 type CommonTransactionReceipt struct {
 	// TransactionHash The hash identifying the transaction
 	TransactionHash *felt.Felt `json:"transaction_hash"`
+	BlockHash       *felt.Felt `json:"block_hash,omitempty"`
```
Comment (rianhughes) [inline rpc/types_transaction_receipt.go]: Why did you add these fields here? They don't belong to the spec https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L2844

File: rpc/types_transaction.go [hunk 7]
```diff
@@ -13,20 +13,26 @@ import (
...
+	Signature             *[]*felt.Felt         `json:"signature,omitempty"`
+	Calldata              *[]*felt.Felt         `json:"calldata,omitempty"`
+	EntryPointSelector    *felt.Felt            `json:"entry_point_selector,omitempty"`
+	CompiledClassHash     *felt.Felt            `json:"compiled_class_hash,omitempty"`
+	ResourceBounds        ResourceBoundsMapping `json:"resource_bounds"`
+	Tip                   U64                   `json:"tip"`
+	PayMasterData         []*felt.Felt          `json:"paymaster_data"`
+	AccountDeploymentData []*felt.Felt          `json:"account_deployment_data"`
+	NonceDataMode         DataAvailabilityMode  `json:"nonce_data_availability_mode"`
+	FeeMode               DataAvailabilityMode  `json:"fee_data_availability_mode"`
```
Comment (rianhughes) [inline rpc/types_transaction.go:35]: Nice

File: rpc/types_transaction_receipt.go (hunk 6 above)
Comment (AryanGodara) [reply rpc/types_transaction_receipt.go]: Both the fields were being returned in the RPC response.
I did not cross-reference the specs. I'll revert this

File: rpc/transaction_test.go (hunk 5 above)
Comment (AryanGodara) [reply rpc/transaction_test.go:235]: I'm getting an error here, where the `TransactionReceipt` object is nil. And I'm not able to figure out how to fix this.
I checked for both mock and testnet, 

Getting: 
```json
actual  : *rpc.TransactionReceiptWithBlockInfo(&rpc.TransactionReceiptWithBlockInfo{TransactionReceipt:
rpc.TransactionReceipt(nil), 
BlockHash:(*felt.Felt)(0x140004e6b40), 
BlockNumber:0xce1f})
```

You see, the transactionReceipt is nil. 
There seems to be an issue while marshalling the data, between the `CommonTransactionReceipt` struct and `TransactionReceipt` interface. But I'm not able to find the exact point where it's going wrong.

Expected :-
```json
expected: rpc.TransactionReceiptWithBlockInfo(rpc.TransactionReceiptWithBlockInfo{TransactionReceipt:rpc.InvokeTransactionReceipt{TransactionHash:(*felt.Felt)(0x1400050fec0), ActualFee:rpc.FeePayment{Amount:(*felt.Felt)(0x1400050fee0), Unit:"FRI"}, ExecutionStatus:"SUCCEEDED", FinalityStatus:"ACCEPTED_ON_L1", Type:"INVOKE", MessagesSent:[]rpc.MsgToL1{}, RevertReason:"", Events:[]rpc.Event{rpc.Event{FromAddress:(*felt.Felt)(0x1400050ff00), Keys:[]*felt.Felt{(*felt.Felt)(0x1400050ffc0)}, Data:[]*felt.Felt{(*felt.Felt)(0x1400050ff20), (*felt.Felt)(0x1400050ff40), (*felt.Felt)(0x1400050ff60), (*felt.Felt)(0x1400050ff80), (*felt.Felt)(0x1400050ffa0)}}, rpc.Event{FromAddress:(*felt.Felt)(0x1400050ffe0), Keys:[]*felt.Felt{(*felt.Felt)(0x140004940a0)}, Data:[]*felt.Felt{(*felt.Felt)(0x14000494020), (*felt.Felt)(0x14000494040), (*felt.Felt)(0x14000494060), (*felt.Felt)(0x14000494080)}}}, ExecutionResources:rpc.ExecutionResources{ComputationResources:rpc.ComputationResources{Steps:5774, MemoryHoles:0, RangeCheckApps:152, PedersenApps:24, PoseidonApps:0, ECOPApps:3, ECDSAApps:0, BitwiseApps:0, KeccakApps:0, SegmentArenaBuiltin:0}, DataAvailability:rpc.DataAvailability{L1Gas:0x0, L1DataGas:0x80}}}, BlockHash:(*felt.Felt)(0x14000494300), BlockNumber:0xce1f})
```

File: rpc/transaction_test.go [hunk 8]
```diff
@@ -118,28 +95,70 @@ func TestTransactionByBlockIdAndIndex(t *testing.T) {
...
 			utils.TestHexToFelt(t, "0x1"),
-			utils.TestHexToFelt(t, "0x13befe6eda920ce4af05a50a67bd808d67eee6ba47bb0892bef2d630eaf1bba"),
+			utils.TestHexToFelt(t, "0x6b74c515944ef1ef630ee1cf08a22e110c39e217fa15554a089182a11f78ed"),
+			utils.TestHexToFelt(t, "0xc844fd57777b0cd7e75c8ea68deec0adf964a6308da7a58de32364b7131cc8"),
+			utils.TestHexToFelt(t, "0x13"),
+			utils.TestHexToFelt(t, "0x41bbf1eff2ac123d9e01004a385329369cbc1c309838562f030b3faa2caa4"),
+			utils.TestHexToFelt(t, "0x54103"),
+			utils.TestHexToFelt(t, "0x7e430a7a59836b5969859b25379c640a8ccb66fb142606d7acb1a5563c2ad9"),
+			utils.TestHexToFelt(t, "0x6600d829"),
+			utils.TestHexToFelt(t, "0x103020400000000000000000000000000000000000000000000000000000000"),
```
Comment (rianhughes) [inline rpc/transaction_test.go:130]: Shouldn't this call data be like this https://sepolia.voyager.online/tx/0x6a4a9c4f1a530f7d6dd7bba9b71f090a70d1e3bbde80998fde11a08aab8b282

File: rpc/transaction_test.go [hunk 9]
```diff
@@ -265,9 +306,9 @@ func TestTransactionReceipt(t *testing.T) {
 			{
 				TxnHash: utils.TestHexToFelt(t, "0x49728601e0bb2f48ce506b0cbd9c0e2a9e50d95858aa41463f46386dca489fd"),
 				ExpectedResp: TransactionReceiptWithBlockInfo{
-					TransactionReceipt: receiptTxnIntegration,
-					BlockNumber:        319132,
-					BlockHash:          utils.TestHexToFelt(t, "0x50e864db6b81ce69fbeb70e6a7284ee2febbb9a2e707415de7adab83525e9cd"),
+					UnknownTransactionReceipt: UnknownTransactionReceipt{receiptTxnIntegration},
```
Comment (rianhughes) [inline rpc/transaction_test.go]: This will need updated as well, but we can leave that for a seperate PR and focus on testnet updates if you'd like

Comment (rianhughes) [reply rpc/transaction_test.go]: Actually, would you mind updating this as well. That way we can mark transaction_test.go as no longer depending on goerli test data

File: rpc/transaction_test.go [hunk 10]
```diff
@@ -304,11 +265,11 @@ func TestTransactionReceipt(t *testing.T) {
 		"mainnet": {},
 		"integration": {
 			{
-				TxnHash: utils.TestHexToFelt(t, "0x49728601e0bb2f48ce506b0cbd9c0e2a9e50d95858aa41463f46386dca489fd"),
+				TxnHash: utils.TestHexToFelt(t, "0xf2f3d50192637e8d5e817363460c39d3a668fe12f117ecedb9749466d8352b"),
```
Comment (rianhughes) [inline rpc/transaction_test.go]: integration is different network than Sepolia, so lets just remove this altogether. Sorry about that!

**Review by rianhughes — APPROVED** [review]
